File: client/src/app/_models/subscriptionHistories/subscriptionHistory.ts
import { Entity } from "src/app/_models/base/entity";

export class SubscriptionHistory extends Entity {
  subscriptionId!: number;
  changedAt!: Date;
  oldStatus!: string;
  newStatus!: string;
  note?: string;

  constructor(init?: Partial<SubscriptionHistory>) {
    super();
    Object.assign(this, init);
  }
}

File: src/MainService.Models/Entities/SubscriptionHistory.cs
namespace MainService.Models.Entities;

public class SubscriptionHistory : BaseEntity
{
    public int UserSubscriptionId { get; set; }
    public UserSubscription UserSubscription { get; set; } = null!;
    public DateTime ChangedAt { get; set; } = DateTime.UtcNow;
    public SubscriptionStatus OldStatus { get; set; }
    public SubscriptionStatus NewStatus { get; set; }
    public string? Note { get; set; }
}